Deleting lower threshold in Aruco pipeline causes photonvision to lock up

Version: dev-v2024.1.1-beta-4.2-8-ge71ce899

When editing the Aruco pipeline settings, deleting the value in the Threshold Min Size (circled input in image) causes the photonvision to become unresponsive. The only way to recover is by restarting photonvision from systemctl or by power cycling the coprocessor.

A similar lock-up occurs when deleting the value of Threshold Max Size, but there are no errors sent to the log in this case.

![image](https://github.com/PhotonVision/photonvision-docs/assets/22793827/467bbabe-be23-44c5-a2c1-1473ea4ea293)

This error repeats in the logs:
```
Jan 08 04:09:02 photonvision java[1591]: [2024-01-08 04:09:02] [VisionModule - VisionRunner - USBFrameProvider - opencv_First Arducam Camera] [ERROR] Exception on loop 1233
Jan 08 04:09:02 photonvision java[1591]: CvException [org.opencv.core.CvException: cv::Exception: OpenCV(4.8.0) /__w/thirdparty-opencv/thirdparty-opencv/opencv/modules/imgproc/src/thresh.cpp:1675: error: (-215:Assertion failed) blockSize % 2 == 1 && blockSize > 1 in function 'adaptiveThreshold'
Jan 08 04:09:02 photonvision java[1591]: ]
Jan 08 04:09:02 photonvision java[1591]:         at org.opencv.imgproc.Imgproc.adaptiveThreshold_0(Native Method)
Jan 08 04:09:02 photonvision java[1591]:         at org.opencv.imgproc.Imgproc.adaptiveThreshold(Imgproc.java:5671)
Jan 08 04:09:02 photonvision java[1591]:         at org.photonvision.vision.pipeline.ArucoPipeline.drawThresholdFrame(ArucoPipeline.java:244)
Jan 08 04:09:02 photonvision java[1591]:         at org.photonvision.vision.pipeline.ArucoPipeline.process(ArucoPipeline.java:150)
Jan 08 04:09:02 photonvision java[1591]:         at org.photonvision.vision.pipeline.ArucoPipeline.process(ArucoPipeline.java:62)
Jan 08 04:09:02 photonvision java[1591]:         at org.photonvision.vision.pipeline.CVPipeline.run(CVPipeline.java:72)
Jan 08 04:09:02 photonvision java[1591]:         at org.photonvision.vision.processes.VisionRunner.update(VisionRunner.java:98)
Jan 08 04:09:02 photonvision java[1591]:         at java.base/java.lang.Thread.run(Thread.java:840)
```

--- a/photon-client/src/components/common/pv-range-slider.vue
+++ b/photon-client/src/components/common/pv-range-slider.vue
@@ -40,11 +40,20 @@ const localValue = computed<[number, number]>({
   }
 });
 
-const changeFromSlot = (v: number, i: number) => {
+const changeFromSlot = (v: string, i: number) => {
+  // v comes in as a string, not a number, for some reason
+  // if v is undefined, take a guess and set it to 0
+  const val = Math.max(props.min, Math.min(parseFloat(v) || 0, props.max));
+
   // localValue.value must be replaced for a reactive change to take place
   const temp = localValue.value;
-  temp[i] = v;
+  temp[i] = val;
   localValue.value = temp;
+};
+
+const checkNumberRange = (v: string): boolean => {
+  const val: number = parseFloat(v);
+  return Boolean(isFinite(val) && val >= props.min && val <= props.max);
 };
 </script>
 
@@ -79,6 +88,7 @@ const changeFromSlot = (v: number, i: number) => {
               :max="max"
               :min="min"
               :step="step"
+              :rules="[checkNumberRange]"
               type="number"
               style="width: 60px"
               @input="(v) => changeFromSlot(v, 0)"
@@ -95,6 +105,7 @@ const changeFromSlot = (v: number, i: number) => {
               :max="max"
               :min="min"
               :step="step"
+              :rules="[checkNumberRange]"
               type="number"
               style="width: 60px"
               @input="(v) => changeFromSlot(v, 1)"

--- a/photon-client/src/components/common/pv-range-slider.vue
+++ b/photon-client/src/components/common/pv-range-slider.vue
@@ -53,7 +53,7 @@ const changeFromSlot = (v: string, i: number) => {
 
 const checkNumberRange = (v: string): boolean => {
   const val: number = parseFloat(v);
-  return Boolean(isFinite(val) && val >= props.min && val <= props.max);
+  return isFinite(val) && val >= props.min && val <= props.max;
 };
 </script>
 

